internal/js/modules/k6/browser/common/frame.go

```diff
@@ -2164,6 +2165,89 @@ func (f *Frame) WaitForURL(urlPattern string, opts *FrameWaitForURLOptions, jsRe
 	return err
 }
 
+// WaitForResponse waits for a response that matches the given URL pattern or predicate function.
+// urlOrPredicate can be a string URL pattern, a regex pattern, or a predicate function.
+func (f *Frame) WaitForResponse(urlOrPredicate string, opts *FrameWaitForResponseOptions, jsRegexChecker JSRegexChecker) (*Response, error) {
+	f.log.Debugf("Frame:WaitForResponse", "fid:%s furl:%q pattern:%s", f.ID(), f.URL(), urlOrPredicate)
+	defer f.log.Debugf("Frame:WaitForResponse:return", "fid:%s furl:%q pattern:%s", f.ID(), f.URL(), urlOrPredicate)
+
+	timeoutCtx, timeoutCancel := context.WithTimeout(f.ctx, opts.Timeout)
+	defer timeoutCancel()
+
+	// Create URL matcher based on the pattern
+	matcher, err := urlMatcher(urlOrPredicate, jsRegexChecker)
+	if err != nil {
+		return nil, fmt.Errorf("parsing URL pattern: %w", err)
+	}
+
+	// Create a channel to receive matching responses
+	responseCh := make(chan *Response, 1)
+	var matcherErr error
+
+	// Create a handler function that will be registered with the page
+	handler := func(event PageOnEvent) error {
+		if event.Response != nil {
+			matched, err := matcher(event.Response.URL())
+			if err != nil {
+				matcherErr = err
+				select {
+				case responseCh <- nil:
+				default:
+				}
+				return nil
+			}
+			if matched {
+				select {
+				case responseCh <- event.Response:
+				default:
+				}
+			}
+		}
+		return nil
+	}
+
+	// Register the handler with the page
+	f.page.eventHandlersMu.Lock()
+	if f.page.eventHandlers[EventPageResponseCalled] == nil {
```

Comment:
I don't think `page. eventHandlers` has been set up to allow for an easy way to remove handlers, and this is why you're having to work with reflect. I'd prefer if a new `responseEventHandler` (or something similar) was created specifically for `page.waitForResponse` where it would be easy to remove the handler by matching against the input url.